src/components/CopyToClipboard/CopyToClipboard.tsx

```diff
@@ -2,23 +2,27 @@
 
 import React from 'react';
 
-import ReactCopyToClipboard from 'react-copy-to-clipboard';
+import {copyText} from '../../utils/copyText';
 
 import type {CopyToClipboardProps, CopyToClipboardStatus} from './types';
 
 const INITIAL_STATUS: CopyToClipboardStatus = 'pending';
 
 export function CopyToClipboard(props: CopyToClipboardProps) {
-    const {children, text, options, timeout, onCopy} = props;
+    const {children, text, timeout, onCopy} = props;
 
+    const textRef = React.useRef(text);
     const [status, setStatus] = React.useState<CopyToClipboardStatus>(INITIAL_STATUS);
 
     const timerIdRef = React.useRef<number>();
 
-    const content = React.useMemo(() => children(status), [children, status]);
+    const content = React.useMemo<React.ReactElement<React.HTMLAttributes<HTMLElement>>>(
+        () => children(status),
+        [children, status],
+    );
```

Comment:
It's no benefit of useMemo at all, children as function typically inline in jsx, so every render is new instance